# Assignment 6

To measure the psychological state of Americans, a psychology researcher (in 1929) devised
three questions put to 3443 Americans from different walks of life. He asked them whether they
were happy. They could answer only **yes** or **no** to each question. By today’s standards the study
is quite simple, and he did no formal statistical analysis of his data. The frequency data (number
of **yes** and **no** responses within each employment category) from this study are in the file
*Americandata.csv*.

---

You need to use the data and answer the question, that is, **Is the type of profession related to the happiness of American people?**
Report the **χ2** statistic and the *statistical significance of test*, in addition to the contingency table.
Also, explain the results and what they mean. What does the value of **χ2** statistic mean? Is it
good or bad? Is the test statistically significant?

In [1]:

import pandas as pd
from scipy.stats import chi2_contingency

df = pd.read_csv('./Americandata.csv')

contingency_table = df.set_index("Happy")

# Perform the chi-square test
chi2, p, dof, expected = chi2_contingency(contingency_table, correction=False)

# Display the contingency table
print("Contingency Table:")
print(contingency_table)

chi_square_statistic = round(chi2, 2)
p_value = round(p,5)

# Display the test results
print(f"\nChi-square statistic: {chi_square_statistic}")
print(f"Degrees of freedom: {dof}")
print(f"P-value (statistical significance): {p_value}")

# Determine statistical significance
if p < 0.05:
    print("The test is statistically significant.")
else:
    print("The test is not statistically significant.")

Contingency Table:
       Students  Laborers  Preachers  Physicians  Housewives  Teachers  \
Happy                                                                    
Yes         390       378         35         159          78       108   
No         1610       122        265          51         122        38   

       Lawyers  Musicians  
Happy                      
Yes         11         31  
No          64         19  

Chi-square statistic: 936.14
Degrees of freedom: 7
P-value (statistical significance): 0.0
The test is statistically significant.


Put your analysis in the empty `markdown` cell below.

Based on this study, the type of profession is related to the happiness of the american people. The chi-square statistic of 936.14 measures how much the observed frequencies differ from the frequencies we would expect if happiness and profession were independent. A larger chi-square statistic indicates a greater difference between the observed and expected frequencies. A chi-square of 936.14 is very high in this case because the degrees of freedom is 7, indicating the variables are related. The test is statistically significant because the p-value is less than 0.05. Therefore, the results indicate a statistically significant association between profession and reported happiness among the American people in the study.